lib/api/CDataFrameTrainBoostedTreeClassifierRunner.cc

```diff
@@ -163,16 +166,34 @@ void CDataFrameTrainBoostedTreeClassifierRunner::writeOneRow(
 
     if (featureImportance != nullptr) {
         featureImportance->shap(
-            row, [&writer](const maths::CTreeShapFeatureImportance::TSizeVec& indices,
-                           const TStrVec& names,
-                           const maths::CTreeShapFeatureImportance::TVectorVec& shap) {
+            row, [&writer, &classValues](
+                     const maths::CTreeShapFeatureImportance::TSizeVec& indices,
+                     const TStrVec& names,
+                     const maths::CTreeShapFeatureImportance::TVectorVec& shap) {
+                writer.Key(FEATURE_IMPORTANCE_FIELD_NAME);
+                writer.StartArray();
                 for (auto i : indices) {
                     if (shap[i].norm() != 0.0) {
-                        writer.Key(names[i]);
-                        // TODO fixme
-                        writer.Double(shap[i](0));
+                        writer.StartObject();
+                        writer.Key(FEATURE_NAME_FIELD_NAME);
+                        writer.String(names[i]);
+                        if (shap[i].size() == 1) {
+                            writer.Key(IMPORTANCE_FIELD_NAME);
+                            writer.Double(shap[i](0));
+                        } else {
+                            double absSum{0.0};
+                            for (int j = 0; j < shap[i].size(); ++j) {
+                                absSum += std::abs(shap[i](j));
+                                writer.Key(classValues[j]);
+                                writer.Double(shap[i](j));
+                            }
+                            writer.Key(IMPORTANCE_FIELD_NAME);
+                            writer.Double(absSum);
```

Comment:
Maybe cleaner to just use the method on the vector type for this
```suggestion
                            writer.Double(shap[i].lpNorm<1>());
```